src/server/app/api/controllers/authController.js

```diff
@@ -1,9 +1,25 @@
 import Boom from 'boom';
 import Bounce from 'bounce';
 import { logErr } from '../../shared/utils';
-import { validateUser } from '../../shared/plugins/auth';
+import { generateToken, validateUser, validateToken } from '../../shared/plugins/auth';
 
 const authController = {
+  async createAuthToken(req, h) {
+    let auth, token;
+    try {
+      if (!req.payload.username) throw new Error("Login requires username.");
+      if (!req.payload.password) throw new Error("Login requires password.");
+      auth = await validateUser(req, req.payload.username, req.payload.password, h);
+      if (!auth.credentials || !auth.isValid) throw new Error("Login not valid.");
+      token = await generateToken(auth, req.payload.username, req.payload.password);
+      if (Bounce.isError(token)) throw new Error(token.message || "No authentication token returned.");
+      const response = h.response(auth);
+      return response;
```

Comment:
I think this function would be a bit more readable if you put strategic whitespace in here. I would suggest putting whitespace around lines 12, 14, and 15.

Also, I think you could just do `return h.response(auth)` at the end to cut a line out.